src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -38,7 +38,11 @@ export default function SavedObjectFactory(es, kbnIndex, Promise, Private, Notif
     let docSource = new DocSource();
 
     // type name for this object, used as the ES-type
-    let type = config.type;
+    const type = config.type;
+
+    self.getDisplayName = function () {
+      return type.charAt(0).toUpperCase() + type.slice(1);
```

Comment:
This is one case where a string-oriented util library would be useful. For now, how about `_.upperFirst(type)`?